smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/ContextHolder.java

```diff
@@ -15,24 +19,29 @@ public ContextHolder(Vertx vertx) {
         this.vertx = vertx;
     }
 
-    public Context getContext() {
-        Context ctx = this.context;
-        if (ctx == null) {
-            synchronized (this) {
-                ctx = this.context;
-                if (ctx == null) {
-                    this.context = ctx = vertx.getOrCreateContext();
-                }
+    public void capture(KafkaReadStream<?, ?> stream) {
+        if (!(stream instanceof KafkaReadStreamImpl)) {
+            throw new IllegalArgumentException("Cannot capture the context - not a KafkaReadStreamImpl");
+        } else {
+            try {
+                Field field = KafkaReadStreamImpl.class.getDeclaredField("context");
+                field.setAccessible(true);
+                context = new Context((io.vertx.core.Context) field.get(stream));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Cannot capture the context", e);
             }
         }
```

Comment:
Will we be living with this as afinal solution or are we going to work the Vert.X to get access to the context properly?

Reply:
That's a good question.
We are going to go with this for now.

There are multiple _things_:

* We need to switch to a blocking rebalance listener - this is done using a hack too (see my other branch - that need to be reworked completely BTW). 
* We only use the Vert.x client to get a `Multi` of Record. That part is not that complicated, so starting to see if we need to remove it completely or not. 
* This code is not changed in Vert.x 3.9 nor Vert.x 4.x (for now). 

I've done a list of improvements to applied in the upstream Vert.x client. Exposing the context needs to be added. Generally, we don't do this in Vert.x as we expect the client creation to be called form an event loop (and so you can capture the context from a creation site), but in our case, it's not the case, and the client creates its own context.